backend/question-service/src/app.ts

```diff
@@ -17,7 +22,10 @@ app.use(cors);
 app.use(bodyParser.json());
 
 // implement routes for API endpoints
-app.use("/api", authMiddleware, router);
+const NODE_ENV = process.env.NODE_ENV || 'development';
+app.use(`/${NODE_ENV}/question`, authMiddleware, router);
```

Comment:
I don't think we should add the service name `question` here. It should be added in the routes path `index.ts`. Furthermore, I believe the reason of having this service name in the endpoint name is to prevent conflicts in `GET /api/health`. So the easy fix is to update the getHealth endpoint to `GET /<NODE_ENV>/questions/health instead. If we add the service name here, then our route becomes like `GET /<NODE_ENV>/question/questions..., which I am not quite comfortable with...

Reply:
Will questions/health conflict with /questions/{questionId}?

Reply:
My bad, you are right, it will...
How about `/${NODE_ENV}/api/health` instead?

Reply:
Ok sorry, ignore all my suggestions, I will stick to your plan. There are endpoints that do not start with `questions`...

Reply:
Can we do `/${NODE_ENV/${SERVICE_NAME/api/xxx`? 
I want to have this endpoint pattern so that we can satisfy what we needed in AWS, as well as prevent confusion and minimize changes needed in our routes pattern. 
For example, for question service it will be:
`/${NODE_ENV}/question/api/questions`
and for user service it will be `/${NODE_ENV}/user/api/users/:userId` for example.
This is helpful for history API because my current route is like:
`GET /api/history`. I don't want it to be `GET /${NODE_ENV}/history/history`, and I don't want to have empty route declaration when I create new route..., so I think `/${NODE_ENV}/history/api/history` will be clearer. Any objection?

Reply:
![image](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g05/assets/7760148/45ffc1a8-9b0e-409a-b0c5-a94b48bc9381)

something like this? i'm okay with it

Reply:
Yes that's what I mean! Thanks! I can help with the changes and update in tests if you don't mind. (That means I will help push the changes to this branch)

Reply:
Definitely 😁

Reply:
Thank you, I will push the changes tonight. Hope it is not too late!